src/Inciter/AMR/node_store.h

```diff
@@ -0,0 +1,245 @@
+#ifndef AMR_node_store_h
+#define AMR_node_store_h
+
+#include <cmath>
+
+#include "types.h"
+#include "tet_store.h"
+
+// TODO: make this have a base class to support multiple generator schemes
+// using the policy design pattern
+namespace AMR {
+
+    class node_store_t
```

Comment:
Yes, lets discuss this tomorrow. Originally I had them merged, but wanted to try and find a discrete way to separate them as it's not clear if everywhere we need AMR (i.e Partitioner.h) we will have a mesh_t